src/lock_file/records_by_name.rs

```diff
@@ -35,6 +35,7 @@ impl HasNameVersion for PypiRecord {
         &self.0.version
     }
 }
+#[allow(refining_impl_trait)]
```

Comment:
Is this needed?

Reply:
The linter demanded it when other traits became pub outside the crate but it seems as if this may have been a compiler bug that is fixed in 1.82.0? https://github.com/rust-lang/rust/pull/129417

Reply:
I removed it. I'm using 1.82.0 but I think CI still is stick'd on 1.81.0. Will see what CI says this build.

Reply:
Looks like it no longer throws a lint error. I had this in our patchset since pixi was building with 1.80 when the lint error would happen.